x-pack/plugins/ml/public/application/capabilities/check_capabilities.ts

```diff
@@ -28,62 +28,80 @@ export function checkGetManagementMlJobsResolver() {
-      _capabilities = capabilities;
-      // the minimum privilege for using ML with a platinum or trial license is being able to get the transforms list.
-      // all other functionality is controlled by the return capabilities object.
-      // if the license is basic (isPlatinumOrTrialLicense === false) then do not redirect,
-      // allow the promise to resolve as the separate license check will redirect then user to
-      // a basic feature
-      if (_capabilities.canGetJobs || isPlatinumOrTrialLicense === false) {
-        return resolve(_capabilities);
-      } else {
+    getCapabilities()
+      .then(({ capabilities, isPlatinumOrTrialLicense }) => {
+        _capabilities = capabilities;
+        // the minimum privilege for using ML with a platinum or trial license is being able to get the transforms list.
+        // all other functionality is controlled by the return capabilities object.
+        // if the license is basic (isPlatinumOrTrialLicense === false) then do not redirect,
+        // allow the promise to resolve as the separate license check will redirect then user to
+        // a basic feature
+        if (_capabilities.canGetJobs || isPlatinumOrTrialLicense === false) {
+          return resolve(_capabilities);
+        } else {
+          window.location.href = '#/access-denied';
+          return reject();
+        }
+      })
+      .catch((e) => {
         window.location.href = '#/access-denied';
         return reject();
-      }
-    });
+      });
   });
 }
 
 export function checkCreateJobsCapabilitiesResolver(): Promise<MlCapabilities> {
   return new Promise((resolve, reject) => {
-    getCapabilities().then(({ capabilities, isPlatinumOrTrialLicense }) => {
-      _capabilities = capabilities;
-      // if the license is basic (isPlatinumOrTrialLicense === false) then do not redirect,
-      // allow the promise to resolve as the separate license check will redirect then user to
-      // a basic feature
-      if (_capabilities.canCreateJob || isPlatinumOrTrialLicense === false) {
-        return resolve(_capabilities);
-      } else {
-        // if the user has no permission to create a job,
-        // redirect them back to the Transforms Management page
+    getCapabilities()
+      .then(({ capabilities, isPlatinumOrTrialLicense }) => {
+        _capabilities = capabilities;
+        // if the license is basic (isPlatinumOrTrialLicense === false) then do not redirect,
+        // allow the promise to resolve as the separate license check will redirect then user to
+        // a basic feature
+        if (_capabilities.canCreateJob || isPlatinumOrTrialLicense === false) {
+          return resolve(_capabilities);
+        } else {
+          // if the user has no permission to create a job,
+          // redirect them back to the Transforms Management page
+          window.location.href = '#/jobs';
+          return reject();
```

Comment:
I presume there is no need to reject the promise when you perform a hard refresh. Would be better to use navigateToUrl instead,